File: tests/fixtures-service/http_clients.ts
// Track G fixtures — HTTP client call sites in TypeScript/JavaScript.
//
// This file is consumed by tests/trackg.ts (Step 3 onward) to verify that
// the TypeScript extractor records ServiceCallDef rows for fetch/axios/etc.

declare const fetch: any;
declare const axios: any;
declare const apiClient: any;
declare const process: any;

// Simple fetch — should yield { protocol: 'http', framework: 'fetch',
// method: 'ANY', rawTarget: '/api/users', normalizedPath: '/api/users' }.
export async function listUsers(): Promise<unknown> {
  return await fetch('/api/users');
}

// fetch with explicit method via options object — method = POST.
export async function createUser(body: unknown): Promise<unknown> {
  return await fetch('/api/users', { method: 'POST', body: JSON.stringify(body) });
}

// axios.<verb> shorthand — method derived from the verb. Caller attribution
// must be `checkout`, not module scope.
export async function checkout(): Promise<unknown> {
  return await axios.post('/checkout', { ok: true });
}

// Generic client.<method>(literalUrl) — should yield framework 'http-client'.
export function fetchOrders(): unknown {
  return apiClient.get('/api/orders');
}

// Template literal with env var — recovers /charge AND envKey PAYMENT_URL.
export async function chargeCustomer(amount: number): Promise<unknown> {
  return await fetch(`${process.env.PAYMENT_URL}/charge`, {
    method: 'POST',
    body: JSON.stringify({ amount }),
  });
}

// Dynamic URL with no literal path — should NOT be recorded.
export async function dynamicUrl(u: string): Promise<unknown> {
  return await fetch(u);
}

// Non-HTTP `.get` (e.g. cache lookup) — first arg is not a string, so dropped.
export function readCache(key: string): unknown {
  return apiClient.get(key);
}

File: tests/fixtures-service/http_clients.py
# Track G fixtures — HTTP client call sites in Python.
import os
import requests
import httpx


def list_users():
    """requests.get('/health') — must yield framework=requests, method=GET."""
    return requests.get("/health")


def create_user(payload):
    """requests.post('/api/users') — POST method."""
    return requests.post("/api/users", json=payload)


def fetch_orders():
    """httpx.get('/api/orders') — framework=httpx, method=GET."""
    return httpx.get("/api/orders")


class CartService:
    def __init__(self, client):
        self.client = client

    def add_item(self, item):
        """client.post('/api/cart/items', json=item) — generic http-client."""
        return self.client.post("/api/cart/items", json=item)


def charge_customer(amount):
    """os.environ['PAYMENT_URL'] + '/charge' — recover envKey=PAYMENT_URL + /charge."""
    return requests.post(os.environ["PAYMENT_URL"] + "/charge", json={"amount": amount})


def dynamic_url(url):
    """No literal path — must NOT be recorded."""
    return requests.get(url)
